src/components/views/elements/ManageIntegsButton.js

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import sdk from '../../../index';
+import SdkConfig from '../../../SdkConfig';
+import ScalarAuthClient from '../../../ScalarAuthClient';
+import ScalarMessaging from '../../../ScalarMessaging';
+import Modal from "../../../Modal";
+import { _t } from '../../../languageHandler';
+import AccessibleButton from './AccessibleButton';
+import TintableSvg from './TintableSvg';
+
+export default class ManageIntegsButton extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            scalarError: null,
+            showIntegrationsError: false,
+        };
+
+        this.onManageIntegrations = this.onManageIntegrations.bind(this);
+        this.onShowIntegrationsError = this.onShowIntegrationsError.bind(this);
+    }
+
+    componentWillMount() {
+        ScalarMessaging.startListening();
+        this.scalarClient = null;
+
+        if (SdkConfig.get().integrations_ui_url && SdkConfig.get().integrations_rest_url) {
+            this.scalarClient = new ScalarAuthClient();
+            this.scalarClient.connect().done(() => {
+                this.forceUpdate();
+            }, (err) => {
+                this.setState({ scalarError: err});
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        ScalarMessaging.stopListening();
+    }
+
+    onManageIntegrations(ev) {
+        ev.preventDefault();
+        const IntegrationsManager = sdk.getComponent("views.settings.IntegrationsManager");
+        Modal.createDialog(IntegrationsManager, {
+            src: (this.scalarClient !== null && this.scalarClient.hasCredentials()) ?
+                this.scalarClient.getScalarInterfaceUrlForRoom(this.props.roomId) :
+                null,
+            onFinished: ()=>{
+                if (this.props.onCancelClick) {
+                    this.props.onCancelClick(ev);
```

Comment:
> for fear of breaking existing expected behaviour.

This property is never set - the entire `if` is redundant and we should absolutely avoid redundant, potentially confusing code.

This means that the behaviour has changed, given you've copied it. Oh! I guess this is the thing where if you close the integrations manager, you also close the room settings. Maybe that wasn't so bizarre after all?